Source/Evolve/WarpXEvolve.cpp

```diff
@@ -96,7 +96,9 @@ WarpX::Evolve (int numsteps)
             FillBoundaryE(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
             FillBoundaryB(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
 #ifdef WARPX_MAG_LLG
-            FillBoundaryM(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
+            if (WarpX::em_solver_medium == MediumForEM::Macroscopic) {
+                FillBoundaryM(guard_cells.ng_alloc_EB, guard_cells.ng_Extra);
+            }
 #endif
```

Comment:
Should this call to FillBoundaryM really have to be wrapped in this if statement?  If so that's fine, but in the future if a non-macroscopic option is implemented, one might forget to take this if statement out.

Reply:
The non-macroscopic option will be implemented in EvolveM.cpp, instead of this function MacroscopicEvolveM.cpp. So I think this should be fine as is.

Reply:
I'm just wondering if there is ever a case where simply always calling FillBoundaryM will hurt anything.
I figured the #ifdef WARPX_MAG_LLG being true was enough to say we cared about M field evolution.